eth/protocols/eth/handshake.go

```diff
@@ -43,14 +43,24 @@ func (p *Peer) Handshake(network uint64, td *big.Int, head common.Hash, genesis
 	var status StatusPacket // safe to read after two values have been received from errc
 
 	go func() {
-		errc <- p2p.Send(p.rw, StatusMsg, &StatusPacket{
-			ProtocolVersion: uint32(p.version),
-			NetworkID:       network,
-			TD:              td,
-			Head:            head,
-			Genesis:         genesis,
-			ForkID:          forkID,
-		})
+		if p.version == ETH68 {
```

Comment:
It would take up less LOC if you make a `StatusPacket`, and then if `p.version == ETH68` set `p.TD = td`, otherwise leave it at `nil`. And then send it.

Reply:
like
```golang
pkt :=  &StatusPacket{
				ProtocolVersion: uint32(p.version),
				NetworkID:       network,
				Head:            head,
				Genesis:         genesis,
				ForkID:          forkID,
			}
if p.version == ETH86{
     pkt.TD =td
}
errc <- p2p.Send(p.rw, pkt)
```